kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardHolderUpdateServiceImpl.java

```diff
@@ -0,0 +1,366 @@
+        return loadedPcardHolders;
+    }
+   
+    /**
+     * Gets the matching record from the custom Procurement Cardholder table.
+     */
+    private ProcurementCardDefault getExistingProcurementCardHolder(String creditCardNumber) {
+                
+        Map<String, String> pkMap = new HashMap<String, String>();
+        pkMap.put(KFSPropertyConstants.CREDIT_CARD_NUMBER, creditCardNumber);
+        ProcurementCardDefault procurementCardHolderDetail = (ProcurementCardDefault) businessObjectService.findByPrimaryKey(ProcurementCardDefault.class, pkMap);
+        
+        return procurementCardHolderDetail;
+    }
+    
+    /**
+     * Check to see if this record is excluded based on the cancel date
+     */
+    private boolean excludeRecord (Date cancelDate) {
+        //exclude records with a cancel date more than 120 days before the go live date 
+        if (ObjectUtils.isNotNull(cancelDate)) {
+            //create go live date
+            Date goLiveDate = Date.valueOf(GO_LIVE_DATE);
+            Timestamp goLiveDateTS = new Timestamp(goLiveDate.getTime());
+            Timestamp cancelDateTS = new Timestamp(cancelDate.getTime());
+            //check difference in days
+            if (goLiveDateTS.after(cancelDateTS) && dateTimeService.dateDiff(cancelDateTS, goLiveDateTS, false) > 150) {
+                return true;
+            }
+            else {
+                return false;
+            }
+        }
+        else {
+            return false;
+        }
+    }
+    
+    /**
+     * Determines the status code from the status text
+     */
+    private String determineStatusCode (String statusText) {
+        //determine valid status code from status text
+        String statusCode;
+        
+        if (statusText.equals(ACTIVE)) {
+            statusCode = STATUS_CODE_A;
+        } else if (statusText.equals(CARD_ACTIVATION)) {
+            statusCode = STATUS_CODE_T;
+        } else if (statusText.equals(CARD_ACTIVATION_MONITORING)) {
+            statusCode = STATUS_CODE_V;
+        } else if (statusText.equals(MONITORING)) {
+            statusCode = STATUS_CODE_M;
+        } else if (statusText.equals(CARD_HELD)) {
+            statusCode = STATUS_CODE_H;
+        } else if (statusText.equals(CHARGEOFF_ACCOUNT)) {
+            statusCode = STATUS_CODE_E;
+        } else if (statusText.equals(NEW)) {
+            statusCode = STATUS_CODE_N;
+        } else if (statusText.equals(OVERLIMIT)) {
+            statusCode = STATUS_CODE_O;
+        } else if (statusText.equals(PENDING)) {
+            statusCode = STATUS_CODE_P;
+        } else if (statusText.equals(TEMPORARY_LOST_STOLEN)) {
+            statusCode = STATUS_CODE_L;
+        } else if (statusText.equals(TEMPORARILY_BLOCKED_STOLEN)) {
+            statusCode = STATUS_CODE_B;
+        } else {
+            statusCode = STATUS_CODE_C;
+        }
+        return statusCode;
```

Comment:
Please add a LOG.debug statement at the end of the method to log the resulting statusCode.